Xamarin.Forms.Platform.Android/Shapes/ShapeRenderer.cs

```diff
@@ -115,7 +115,15 @@ void UpdateStrokeThickness()
 
         void UpdateStrokeDashArray()
         {
-            Control.UpdateStrokeDashArray(Element.StrokeDashArray.ToArray());
+            var array = Element.StrokeDashArray.ToArray();
+
+            double thickness = Element.StrokeThickness;
+            var dashArray = new float[array.Length];
+
+            for (int i = 0; i < array.Length; i++)
+                dashArray[i] = (float)(thickness * array[i]);
```

Comment:
Right now, the `ToArray` extension is creating an array which is then being copied again with some modifications (a cast to `float` and a scaling operation) - that means allocating an entire array which is immediately discarded. Is there any reason the cast and scaling can't be applied in the `ToArray` method, so we only allocate one extra array?